### ISSUE TITLE
use crochet's overlay feature for customizing  

### ISSUE THREAD

**Issue by MoritzFago** [hughobrien/zfs-remote-mirror#3]
Crochet has a overlay feature to allow for custom fstab and co, so one could use that to save the mounting of the filesystem.

```
# OVERLAY FILES
#
# Most customization simply consists of putting extra
# files onto the image.
#
# After the board logic has populated the FreeBSD system, the contents
# of ${BOARDDIR}/overlay, ${TOPDIR}/overlay, and ${WORKDIR}/overlay
# are copied on top of the image in that order.  For example, you can
# put a custom passwd or fstab file into ${TOPDIR}/overlay/etc and it
# will be picked up automatically.  Similarly, if you need to
# dynamically create a certain file, you can do that here (config.sh
# is a /bin/sh file and you can use arbitrary shell logic) and put the
# result into ${WORKDIR}/overlay
```

**Comment by hughobrien** [comment]
Yup, would definitely shorten the configuration process. There's a few parts that I think still have to be set manually, either before writing the image or after booting it. Not sure if there's a preference either way.
- hostname
- keymap
- SSH CA signing key
- nameservers
- entropy

Any suggestions for those?

**Comment by MoritzFago** [comment]
Why should you have to set these parts manually? If i get this right, there's no problem with putting all the needed files in the right path under `${WORKDIR}/overlay`.
So for example the rc.conf: `${WORKDIR}/overlay/etc/rc.conf`.
Same should work for all the other files. Of course you still have to prepare the files, but there isn't any need for mounting the image before writing.

**Comment by hughobrien** [comment]
>  Of course you still have to prepare the files

This was what I didn't understand. This is a better way - I'll include it in an update that includes 10.3 and keeps IPv6. Thank you.

I'm also working on an easier way to test the build and create packages using QEMU.
